simple-gotracing/http2_trace_uprobe/bpf_program.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+// This does not work for Golang programs built with 1.17 or newer toolchain. Go 1.17 uses a register-based calling
+// convention for which the BPF code here cannot handle.
+const bpfProgram = `
+#include <uapi/linux/ptrace.h>
+
+#define HEADER_FIELD_STR_SIZE 128
+#define MAX_HEADER_COUNT 64
+
+BPF_PERF_OUTPUT(go_http2_header_events);
+
+struct header_field_t {
+  int32_t size;
+  char msg[HEADER_FIELD_STR_SIZE];
+};
+
+struct go_grpc_http2_header_event_t {
+  struct header_field_t name;
+  struct header_field_t value;
+};
+
+// This matches golang string's memory layout, help reading data from golang string object.
```

Comment:
"This matches the golang string object memory layout. Used to help read golang string objects in BPF code."